### PR TITLE
update GH templates' labels

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Updating rules for using "help wanted" labels, see #9519

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/code_improvement.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name: Code improvement
 about: Suggest a code improvement, i.e. refactoring, deprecation, etc.
 title: ''
-labels: enhancement, help wanted, refactors / code health
+labels: enhancement, refactors / code health
```
Comment (carmocca) [inline .github/ISSUE_TEMPLATE/code_improvement.md:5]: why does this one have `enhancement` too?

Comment (daniellepintz) [reply .github/ISSUE_TEMPLATE/code_improvement.md:5]: I think the idea is code enhancement, but we could also remove

**Comment by daniellepintz** [comment]
@Borda this change looks good to me! 
One comment about the workflow:

>when good first issue remove help wanted

This doesn't seem necessary, an issue could definitely fall into both categories right?

**Review by daniellepintz — APPROVED** [review]

**Comment by Borda** [comment]
> I'm wondering do we even need the `let's do it` label then? both signify we have decided to implement the issue right?

that is a good point we can simplify it, when we add issue to the action bucker (add a milestone) we assume it shall be implemented

**Review by tchaton — APPROVED** [review]
LGMT !

**Comment by daniellepintz** [comment]
@Borda are you planning on publishing this PR?

**Comment by Borda** [comment]
> @Borda are you planning on publishing this PR?

yes, just need to finish the label change action

**Comment by daniellepintz** [comment]
@Borda thinking about it more, in my opinion even the bug issue shouldn't have the `help wanted` tag. The reason being bugs are usually something one of the core maintainers investigates/fixes, rather than a community member, and I see the `help wanted` tag as signaling to the community that this issue is open and endorsed by a maintainer to work on. Just stating my opinion, feel free to disagree :)

**Comment by Borda** [comment]
I would keep it as mostly all bugs we want to fix and if a core member files a bug report for himself he can simply don't add this label... also when he assigns it himself the label is removed

**Comment by daniellepintz** [comment]
I guess my worry is people file bug reports that aren't actually bugs, something like https://github.com/PyTorchLightning/pytorch-lightning/issues/9286, and then if the help wanted label is added someone might start working on it even if it's not something we want to fix. By making the help wanted label something that is always added manually it makes it very reliable. But I understand your point of view, and maybe the example I gave was an anomaly

**Comment by Borda** [comment]
> I guess my worry is people file bug reports that aren't actually bugs

true but still you need someone to replicate it and why community could hot help here? :]
the label is "help wanted" not "fix me"... anyway lets start step by step and we can adjust some processes later

**Comment by Borda** [comment]
ok, seems that the automation is not so trivial, so let's first merge template update and do CI in the next round :rabbit:

**Review by awaelchli — APPROVED** [review]

**Pull request merged**